2020-11-14 00:53:51.476 | DEBUG    | aria2p.api:pause:523 - Failed to pause download 0000000000000001
2020-11-14 00:53:51.477 | TRACE    | aria2p.api:pause:524 - GID#0000000000000001 cannot be paused now
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "<string>", line 8, in <module>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1554, in serve
    SlaveGateway(io=io, id=id, _startcount=2).serve()
    │               │      └ 'gw1-slave'
    │               └ <execnet.gateway_base.Popen2IO object at 0x7fed0a01dd30>
    └ <class 'execnet.gateway_base.SlaveGateway'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1060, in serve
    self._execpool.integrate_as_primary_thread()
    │    │         └ <function WorkerPool.integrate_as_primary_thread at 0x7fed09fd4730>
    │    └ <execnet.gateway_base.WorkerPool object at 0x7fed09fc63c8>
    └ <execnet.gateway_base.SlaveGateway object at 0x7fed09fc62e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 267, in integrate_as_primary_thread
    self._perform_spawn(reply)
    │    │              └ <execnet.gateway_base.Reply object at 0x7fed09f6f860>
    │    └ <function WorkerPool._perform_spawn at 0x7fed09fd48c8>
    └ <execnet.gateway_base.WorkerPool object at 0x7fed09fc63c8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 285, in _perform_spawn
    reply.run()
    │     └ <function Reply.run at 0x7fed09fd4620>
    └ <execnet.gateway_base.Reply object at 0x7fed09f6f860>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 220, in run
    self._result = func(*args, **kwargs)
    │              │     │       └ {}
    │              │     └ ((<Channel id=1 open>, ('"""\n    This module is executed in remote subprocesses and helps to\n    control a remote testing s...
    │              └ <bound method SlaveGateway.executetask of <execnet.gateway_base.SlaveGateway object at 0x7fed09fc62e8>>
    └ <execnet.gateway_base.Reply object at 0x7fed09f6f860>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1084, in executetask
    do_exec(co, loc)  # noqa
    │       │   └ {'channel': <Channel id=1 open>, '__name__': '__channelexec__', '__builtins__': {'__name__': 'builtins', '__doc__': "Built-in...
    │       └ <code object <module> at 0x7fed09f71ed0, file "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/...
    └ <built-in function exec>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 237, in <module>
    config.hook.pytest_cmdline_main(config=config)
    │      │    │                          └ <_pytest.config.Config object at 0x7fed08bb8320>
    │      │    └ <_HookCaller 'pytest_cmdline_main'>
    │      └ <pluggy.hooks._HookRelay object at 0x7fed08c44e48>
    └ <_pytest.config.Config object at 0x7fed08bb8320>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'config': <_pytest.config.Config object at 0x7fed08bb8320>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7fed08ea8158>
           │    │         │     └ <_HookCaller 'pytest_cmdline_main'>
           │    │         └ <_HookCaller 'pytest_cmdline_main'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7fed096a5710>>
           └ <_HookCaller 'pytest_cmdline_main'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'config': <_pytest.config.Config object at 0x7fed08bb8320>}
           │    │               │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-...
           │    │               └ <_HookCaller 'pytest_cmdline_main'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7fed08bae7b8>
           └ <_pytest.config.PytestPluginManager object at 0x7fed096a5710>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7fed08ba8d30>
                │    │    │    │                     └ <_HookCaller 'pytest_cmdline_main'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7fed08ba8d30>
                └ <_HookCaller 'pytest_cmdline_main'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<_pytest.config.Config object at 0x7fed08bb8320>]
          │         └ <function pytest_cmdline_main at 0x7fed08c957b8>
          └ <HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-p...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 306, in pytest_cmdline_main
    return wrap_session(config, _main)
           │            │       └ <function _main at 0x7fed08c95950>
           │            └ <_pytest.config.Config object at 0x7fed08bb8320>
           └ <function wrap_session at 0x7fed08c95840>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 257, in wrap_session
    session.exitstatus = doit(config, session) or 0
    │       │            │    │       └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=12 testscollected=23450>
    │       │            │    └ <_pytest.config.Config object at 0x7fed08bb8320>
    │       │            └ <function _main at 0x7fed08c95950>
    │       └ <ExitCode.OK: 0>
    └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=12 testscollected=23450>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 313, in _main
    config.hook.pytest_runtestloop(session=session)
    │      │    │                          └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=12 testscollected=23450>
    │      │    └ <_HookCaller 'pytest_runtestloop'>
    │      └ <pluggy.hooks._HookRelay object at 0x7fed08c44e48>
    └ <_pytest.config.Config object at 0x7fed08bb8320>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'session': <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=12 testscollected=23450>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7fed08ea8158>
           │    │         │     └ <_HookCaller 'pytest_runtestloop'>
           │    │         └ <_HookCaller 'pytest_runtestloop'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7fed096a5710>>
           └ <_HookCaller 'pytest_runtestloop'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'session': <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=12 testscollected=23450>}
           │    │               │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-...
           │    │               └ <_HookCaller 'pytest_runtestloop'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7fed08bae7b8>
           └ <_pytest.config.PytestPluginManager object at 0x7fed096a5710>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7fed08bb3f60>
                │    │    │    │                     └ <_HookCaller 'pytest_runtestloop'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7fed08bb3f60>
                └ <_HookCaller 'pytest_runtestloop'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Session tests exitstatus=<ExitCode.OK: 0> testsfailed=12 testscollected=23450>]
          │         └ <bound method WorkerInteractor.pytest_runtestloop of <__channelexec__.WorkerInteractor object at 0x7fed0389f6a0>>
          └ <HookImpl plugin_name='140655943349920', plugin=<__channelexec__.WorkerInteractor object at 0x7fed0389f6a0>>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 70, in pytest_runtestloop
    self.run_one_test(torun)
    │    │            └ [16065, 16066, 16067, 16068, 16069, 16070, 16071, 16072, 16073, 16074, 16075, 16076, 16077, 16078, 16079, 16080, 16081, 16082...
    │    └ <function WorkerInteractor.run_one_test at 0x7fed08bae268>
    └ <__channelexec__.WorkerInteractor object at 0x7fed0389f6a0>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 87, in run_one_test
    self.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
    │    │      │    │                            │              └ <Function test_add_torrent_subcommand[29]>
    │    │      │    │                            └ <Function test_pause_subcommand_already_paused[28]>
    │    │      │    └ <_HookCaller 'pytest_runtest_protocol'>
    │    │      └ <pluggy.hooks._HookRelay object at 0x7fed08c44e48>
    │    └ <_pytest.config.Config object at 0x7fed08bb8320>
    └ <__channelexec__.WorkerInteractor object at 0x7fed0389f6a0>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'item': <Function test_pause_subcommand_already_paused[28]>, 'nextitem': <Function test_add_torrent_subcommand[29]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7fed08ea8158>
           │    │         │     └ <_HookCaller 'pytest_runtest_protocol'>
           │    │         └ <_HookCaller 'pytest_runtest_protocol'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7fed096a5710>>
           └ <_HookCaller 'pytest_runtest_protocol'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'item': <Function test_pause_subcommand_already_paused[28]>, 'nextitem': <Function test_add_torrent_subcommand[29]>}
           │    │               │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_runtest_protocol'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7fed08bae7b8>
           └ <_pytest.config.PytestPluginManager object at 0x7fed096a5710>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7fed08bb3e10>
                │    │    │    │                     └ <_HookCaller 'pytest_runtest_protocol'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7fed08bb3e10>
                └ <_HookCaller 'pytest_runtest_protocol'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_pause_subcommand_already_paused[28]>, <Function test_add_torrent_subcommand[29]>]
          │         └ <function pytest_runtest_protocol at 0x7fed08c8e1e0>
          └ <HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQ...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 110, in pytest_runtest_protocol
    runtestprotocol(item, nextitem=nextitem)
    │               │              └ <Function test_add_torrent_subcommand[29]>
    │               └ <Function test_pause_subcommand_already_paused[28]>
    └ <function runtestprotocol at 0x7fed08c8e2f0>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 127, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
    │       │      │               │             └ True
    │       │      │               └ <Function test_pause_subcommand_already_paused[28]>
    │       │      └ <function call_and_report at 0x7fed08c8e620>
    │       └ <method 'append' of 'list' objects>
    └ [<TestReport 'test_cli.py::test_pause_subcommand_already_paused[28]' when='setup' outcome='passed'>]
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 216, in call_and_report
    call = call_runtest_hook(item, when, **kwds)
           │                 │     │       └ {}
           │                 │     └ 'call'
           │                 └ <Function test_pause_subcommand_already_paused[28]>
           └ <function call_runtest_hook at 0x7fed08c8e840>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 256, in call_runtest_hook
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            │          │       │           │             └ (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)
            │          │       │           └ 'call'
            │          │       └ {}
            │          └ <Function test_pause_subcommand_already_paused[28]>
            └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 310, in from_call
    result = func()  # type: Optional[TResult]
             └ <function call_runtest_hook.<locals>.<lambda> at 0x7fecfa605ae8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 256, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            │          │       └ {}
            │          └ <Function test_pause_subcommand_already_paused[28]>
            └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'item': <Function test_pause_subcommand_already_paused[28]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7fed08ea8158>
           │    │         │     └ <_HookCaller 'pytest_runtest_call'>
           │    │         └ <_HookCaller 'pytest_runtest_call'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7fed096a5710>>
           └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'item': <Function test_pause_subcommand_already_paused[28]>}
           │    │               │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_runtest_call'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7fed08bae7b8>
           └ <_pytest.config.PytestPluginManager object at 0x7fed096a5710>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7fed08bb3be0>
                │    │    │    │                     └ <_HookCaller 'pytest_runtest_call'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': False, 'historic': False}
                │    └ <pluggy.hooks.HookSpec object at 0x7fed08bb3be0>
                └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_pause_subcommand_already_paused[28]>]
          │         └ <function pytest_runtest_call at 0x7fed08c8e488>
          └ <HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQ...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 163, in pytest_runtest_call
    item.runtest()
    │    └ <function Function.runtest at 0x7fed08c540d0>
    └ <Function test_pause_subcommand_already_paused[28]>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/python.py", line 1627, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
    │    │                                   └ <Function test_pause_subcommand_already_paused[28]>
    │    └ <property object at 0x7fed08ce6778>
    └ <Function test_pause_subcommand_already_paused[28]>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'pyfuncitem': <Function test_pause_subcommand_already_paused[28]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7fed08ea8158>
           │    │         │     └ <_HookCaller 'pytest_pyfunc_call'>
           │    │         └ <_HookCaller 'pytest_pyfunc_call'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7fed096a5710>>
           └ <_HookCaller 'pytest_pyfunc_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'pyfuncitem': <Function test_pause_subcommand_already_paused[28]>}
           │    │               │     └ [<HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_pyfunc_call'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7fed08bae7b8>
           └ <_pytest.config.PytestPluginManager object at 0x7fed096a5710>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7fed08bb3940>
                │    │    │    │                     └ <_HookCaller 'pytest_pyfunc_call'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7fed08bb3940>
                └ <_HookCaller 'pytest_pyfunc_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_pause_subcommand_already_paused[28]>]
          │         └ <function pytest_pyfunc_call at 0x7fed08c482f0>
          └ <HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQ...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/python.py", line 184, in pytest_pyfunc_call
    result = testfunction(**testargs)
             │              └ {'tmp_path': PosixPath('/tmp/pytest-of-pawamoy/pytest-66/popen-gw1/test_pause_subcommand_already_0'), 'port': 15449, 'capsys'...
             └ <function test_pause_subcommand_already_paused at 0x7fed022a9d90>

  File "/media/data/dev/pawamoy/aria2p/tests/test_cli.py", line 149, in test_pause_subcommand_already_paused
    assert cli.subcommand_pause(server.api, ["0000000000000001", "0000000000000002"]) == 1
           │   │                │      └ API(Client(host='http://localhost', port=15449, secret='********'))
           │   │                └ <tests.conftest._Aria2Server object at 0x7fecfa6bd550>
           │   └ <function subcommand_pause at 0x7fed022a7d90>
           └ <module 'aria2p.cli' from '/media/data/dev/pawamoy/aria2p/src/aria2p/cli.py'>

  File "/media/data/dev/pawamoy/aria2p/src/aria2p/cli.py", line 610, in subcommand_pause
    result = api.pause(downloads, force=force)
             │   │     │                └ False
             │   │     └ [<aria2p.downloads.Download object at 0x7fecfa6bdcf8>, <aria2p.downloads.Download object at 0x7fecfa6bd240>]
             │   └ <function API.pause at 0x7fed0388a268>
             └ API(Client(host='http://localhost', port=15449, secret='********'))

> File "/media/data/dev/pawamoy/aria2p/src/aria2p/api.py", line 521, in pause
    pause_func(download.gid)
    │          │        └ <property object at 0x7fed038783b8>
    │          └ <aria2p.downloads.Download object at 0x7fecfa6bdcf8>
    └ <bound method Client.pause of Client(host='http://localhost', port=15449, secret='********')>

  File "/media/data/dev/pawamoy/aria2p/src/aria2p/client.py", line 672, in pause
    return self.call(self.PAUSE, [gid])
           │    │    │    │       └ '0000000000000001'
           │    │    │    └ 'aria2.pause'
           │    │    └ Client(host='http://localhost', port=15449, secret='********')
           │    └ <function Client.call at 0x7fed03971c80>
           └ Client(host='http://localhost', port=15449, secret='********')

  File "/media/data/dev/pawamoy/aria2p/src/aria2p/client.py", line 261, in call
    return self.res_or_raise(self.post(self.get_payload(method, params, msg_id=msg_id)))
           │    │            │    │    │    │           │       │              └ None
           │    │            │    │    │    │           │       └ ['0000000000000001']
           │    │            │    │    │    │           └ 'aria2.pause'
           │    │            │    │    │    └ <staticmethod object at 0x7fed0394c630>
           │    │            │    │    └ Client(host='http://localhost', port=15449, secret='********')
           │    │            │    └ <function Client.post at 0x7fed03971ea0>
           │    │            └ Client(host='http://localhost', port=15449, secret='********')
           │    └ <staticmethod object at 0x7fed0394c6a0>
           └ Client(host='http://localhost', port=15449, secret='********')

  File "/media/data/dev/pawamoy/aria2p/src/aria2p/client.py", line 391, in res_or_raise
    raise Client.response_as_exception(response)
          │      │                     └ {'id': -1, 'jsonrpc': '2.0', 'error': {'code': 1, 'message': 'GID#0000000000000001 cannot be paused now'}}
          │      └ <staticmethod object at 0x7fed039877b8>
          └ <class 'aria2p.client.Client'>

aria2p.client.ClientException: GID#0000000000000001 cannot be paused now
2020-11-14 00:53:51.539 | DEBUG    | aria2p.api:pause:523 - Failed to pause download 0000000000000002
2020-11-14 00:53:51.539 | TRACE    | aria2p.api:pause:524 - GID#0000000000000002 cannot be paused now
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "<string>", line 8, in <module>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1554, in serve
    SlaveGateway(io=io, id=id, _startcount=2).serve()
    │               │      └ 'gw1-slave'
    │               └ <execnet.gateway_base.Popen2IO object at 0x7fed0a01dd30>
    └ <class 'execnet.gateway_base.SlaveGateway'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1060, in serve
    self._execpool.integrate_as_primary_thread()
    │    │         └ <function WorkerPool.integrate_as_primary_thread at 0x7fed09fd4730>
    │    └ <execnet.gateway_base.WorkerPool object at 0x7fed09fc63c8>
    └ <execnet.gateway_base.SlaveGateway object at 0x7fed09fc62e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 267, in integrate_as_primary_thread
    self._perform_spawn(reply)
    │    │              └ <execnet.gateway_base.Reply object at 0x7fed09f6f860>
    │    └ <function WorkerPool._perform_spawn at 0x7fed09fd48c8>
    └ <execnet.gateway_base.WorkerPool object at 0x7fed09fc63c8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 285, in _perform_spawn
    reply.run()
    │     └ <function Reply.run at 0x7fed09fd4620>
    └ <execnet.gateway_base.Reply object at 0x7fed09f6f860>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 220, in run
    self._result = func(*args, **kwargs)
    │              │     │       └ {}
    │              │     └ ((<Channel id=1 open>, ('"""\n    This module is executed in remote subprocesses and helps to\n    control a remote testing s...
    │              └ <bound method SlaveGateway.executetask of <execnet.gateway_base.SlaveGateway object at 0x7fed09fc62e8>>
    └ <execnet.gateway_base.Reply object at 0x7fed09f6f860>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1084, in executetask
    do_exec(co, loc)  # noqa
    │       │   └ {'channel': <Channel id=1 open>, '__name__': '__channelexec__', '__builtins__': {'__name__': 'builtins', '__doc__': "Built-in...
    │       └ <code object <module> at 0x7fed09f71ed0, file "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/...
    └ <built-in function exec>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 237, in <module>
    config.hook.pytest_cmdline_main(config=config)
    │      │    │                          └ <_pytest.config.Config object at 0x7fed08bb8320>
    │      │    └ <_HookCaller 'pytest_cmdline_main'>
    │      └ <pluggy.hooks._HookRelay object at 0x7fed08c44e48>
    └ <_pytest.config.Config object at 0x7fed08bb8320>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'config': <_pytest.config.Config object at 0x7fed08bb8320>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7fed08ea8158>
           │    │         │     └ <_HookCaller 'pytest_cmdline_main'>
           │    │         └ <_HookCaller 'pytest_cmdline_main'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7fed096a5710>>
           └ <_HookCaller 'pytest_cmdline_main'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'config': <_pytest.config.Config object at 0x7fed08bb8320>}
           │    │               │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-...
           │    │               └ <_HookCaller 'pytest_cmdline_main'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7fed08bae7b8>
           └ <_pytest.config.PytestPluginManager object at 0x7fed096a5710>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7fed08ba8d30>
                │    │    │    │                     └ <_HookCaller 'pytest_cmdline_main'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7fed08ba8d30>
                └ <_HookCaller 'pytest_cmdline_main'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<_pytest.config.Config object at 0x7fed08bb8320>]
          │         └ <function pytest_cmdline_main at 0x7fed08c957b8>
          └ <HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-p...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 306, in pytest_cmdline_main
    return wrap_session(config, _main)
           │            │       └ <function _main at 0x7fed08c95950>
           │            └ <_pytest.config.Config object at 0x7fed08bb8320>
           └ <function wrap_session at 0x7fed08c95840>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 257, in wrap_session
    session.exitstatus = doit(config, session) or 0
    │       │            │    │       └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=12 testscollected=23450>
    │       │            │    └ <_pytest.config.Config object at 0x7fed08bb8320>
    │       │            └ <function _main at 0x7fed08c95950>
    │       └ <ExitCode.OK: 0>
    └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=12 testscollected=23450>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 313, in _main
    config.hook.pytest_runtestloop(session=session)
    │      │    │                          └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=12 testscollected=23450>
    │      │    └ <_HookCaller 'pytest_runtestloop'>
    │      └ <pluggy.hooks._HookRelay object at 0x7fed08c44e48>
    └ <_pytest.config.Config object at 0x7fed08bb8320>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'session': <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=12 testscollected=23450>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7fed08ea8158>
           │    │         │     └ <_HookCaller 'pytest_runtestloop'>
           │    │         └ <_HookCaller 'pytest_runtestloop'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7fed096a5710>>
           └ <_HookCaller 'pytest_runtestloop'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'session': <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=12 testscollected=23450>}
           │    │               │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-...
           │    │               └ <_HookCaller 'pytest_runtestloop'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7fed08bae7b8>
           └ <_pytest.config.PytestPluginManager object at 0x7fed096a5710>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7fed08bb3f60>
                │    │    │    │                     └ <_HookCaller 'pytest_runtestloop'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7fed08bb3f60>
                └ <_HookCaller 'pytest_runtestloop'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Session tests exitstatus=<ExitCode.OK: 0> testsfailed=12 testscollected=23450>]
          │         └ <bound method WorkerInteractor.pytest_runtestloop of <__channelexec__.WorkerInteractor object at 0x7fed0389f6a0>>
          └ <HookImpl plugin_name='140655943349920', plugin=<__channelexec__.WorkerInteractor object at 0x7fed0389f6a0>>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 70, in pytest_runtestloop
    self.run_one_test(torun)
    │    │            └ [16065, 16066, 16067, 16068, 16069, 16070, 16071, 16072, 16073, 16074, 16075, 16076, 16077, 16078, 16079, 16080, 16081, 16082...
    │    └ <function WorkerInteractor.run_one_test at 0x7fed08bae268>
    └ <__channelexec__.WorkerInteractor object at 0x7fed0389f6a0>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 87, in run_one_test
    self.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
    │    │      │    │                            │              └ <Function test_add_torrent_subcommand[29]>
    │    │      │    │                            └ <Function test_pause_subcommand_already_paused[28]>
    │    │      │    └ <_HookCaller 'pytest_runtest_protocol'>
    │    │      └ <pluggy.hooks._HookRelay object at 0x7fed08c44e48>
    │    └ <_pytest.config.Config object at 0x7fed08bb8320>
    └ <__channelexec__.WorkerInteractor object at 0x7fed0389f6a0>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'item': <Function test_pause_subcommand_already_paused[28]>, 'nextitem': <Function test_add_torrent_subcommand[29]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7fed08ea8158>
           │    │         │     └ <_HookCaller 'pytest_runtest_protocol'>
           │    │         └ <_HookCaller 'pytest_runtest_protocol'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7fed096a5710>>
           └ <_HookCaller 'pytest_runtest_protocol'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'item': <Function test_pause_subcommand_already_paused[28]>, 'nextitem': <Function test_add_torrent_subcommand[29]>}
           │    │               │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_runtest_protocol'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7fed08bae7b8>
           └ <_pytest.config.PytestPluginManager object at 0x7fed096a5710>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7fed08bb3e10>
                │    │    │    │                     └ <_HookCaller 'pytest_runtest_protocol'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7fed08bb3e10>
                └ <_HookCaller 'pytest_runtest_protocol'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_pause_subcommand_already_paused[28]>, <Function test_add_torrent_subcommand[29]>]
          │         └ <function pytest_runtest_protocol at 0x7fed08c8e1e0>
          └ <HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQ...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 110, in pytest_runtest_protocol
    runtestprotocol(item, nextitem=nextitem)
    │               │              └ <Function test_add_torrent_subcommand[29]>
    │               └ <Function test_pause_subcommand_already_paused[28]>
    └ <function runtestprotocol at 0x7fed08c8e2f0>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 127, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
    │       │      │               │             └ True
    │       │      │               └ <Function test_pause_subcommand_already_paused[28]>
    │       │      └ <function call_and_report at 0x7fed08c8e620>
    │       └ <method 'append' of 'list' objects>
    └ [<TestReport 'test_cli.py::test_pause_subcommand_already_paused[28]' when='setup' outcome='passed'>]
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 216, in call_and_report
    call = call_runtest_hook(item, when, **kwds)
           │                 │     │       └ {}
           │                 │     └ 'call'
           │                 └ <Function test_pause_subcommand_already_paused[28]>
           └ <function call_runtest_hook at 0x7fed08c8e840>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 256, in call_runtest_hook
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            │          │       │           │             └ (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)
            │          │       │           └ 'call'
            │          │       └ {}
            │          └ <Function test_pause_subcommand_already_paused[28]>
            └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 310, in from_call
    result = func()  # type: Optional[TResult]
             └ <function call_runtest_hook.<locals>.<lambda> at 0x7fecfa605ae8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 256, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            │          │       └ {}
            │          └ <Function test_pause_subcommand_already_paused[28]>
            └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'item': <Function test_pause_subcommand_already_paused[28]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7fed08ea8158>
           │    │         │     └ <_HookCaller 'pytest_runtest_call'>
           │    │         └ <_HookCaller 'pytest_runtest_call'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7fed096a5710>>
           └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'item': <Function test_pause_subcommand_already_paused[28]>}
           │    │               │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_runtest_call'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7fed08bae7b8>
           └ <_pytest.config.PytestPluginManager object at 0x7fed096a5710>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7fed08bb3be0>
                │    │    │    │                     └ <_HookCaller 'pytest_runtest_call'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': False, 'historic': False}
                │    └ <pluggy.hooks.HookSpec object at 0x7fed08bb3be0>
                └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_pause_subcommand_already_paused[28]>]
          │         └ <function pytest_runtest_call at 0x7fed08c8e488>
          └ <HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQ...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 163, in pytest_runtest_call
    item.runtest()
    │    └ <function Function.runtest at 0x7fed08c540d0>
    └ <Function test_pause_subcommand_already_paused[28]>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/python.py", line 1627, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
    │    │                                   └ <Function test_pause_subcommand_already_paused[28]>
    │    └ <property object at 0x7fed08ce6778>
    └ <Function test_pause_subcommand_already_paused[28]>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'pyfuncitem': <Function test_pause_subcommand_already_paused[28]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7fed08ea8158>
           │    │         │     └ <_HookCaller 'pytest_pyfunc_call'>
           │    │         └ <_HookCaller 'pytest_pyfunc_call'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7fed096a5710>>
           └ <_HookCaller 'pytest_pyfunc_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'pyfuncitem': <Function test_pause_subcommand_already_paused[28]>}
           │    │               │     └ [<HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_pyfunc_call'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7fed08bae7b8>
           └ <_pytest.config.PytestPluginManager object at 0x7fed096a5710>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7fed08bb3940>
                │    │    │    │                     └ <_HookCaller 'pytest_pyfunc_call'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7fed08bb3940>
                └ <_HookCaller 'pytest_pyfunc_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_pause_subcommand_already_paused[28]>]
          │         └ <function pytest_pyfunc_call at 0x7fed08c482f0>
          └ <HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQ...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/python.py", line 184, in pytest_pyfunc_call
    result = testfunction(**testargs)
             │              └ {'tmp_path': PosixPath('/tmp/pytest-of-pawamoy/pytest-66/popen-gw1/test_pause_subcommand_already_0'), 'port': 15449, 'capsys'...
             └ <function test_pause_subcommand_already_paused at 0x7fed022a9d90>

  File "/media/data/dev/pawamoy/aria2p/tests/test_cli.py", line 149, in test_pause_subcommand_already_paused
    assert cli.subcommand_pause(server.api, ["0000000000000001", "0000000000000002"]) == 1
           │   │                │      └ API(Client(host='http://localhost', port=15449, secret='********'))
           │   │                └ <tests.conftest._Aria2Server object at 0x7fecfa6bd550>
           │   └ <function subcommand_pause at 0x7fed022a7d90>
           └ <module 'aria2p.cli' from '/media/data/dev/pawamoy/aria2p/src/aria2p/cli.py'>

  File "/media/data/dev/pawamoy/aria2p/src/aria2p/cli.py", line 610, in subcommand_pause
    result = api.pause(downloads, force=force)
             │   │     │                └ False
             │   │     └ [<aria2p.downloads.Download object at 0x7fecfa6bdcf8>, <aria2p.downloads.Download object at 0x7fecfa6bd240>]
             │   └ <function API.pause at 0x7fed0388a268>
             └ API(Client(host='http://localhost', port=15449, secret='********'))

> File "/media/data/dev/pawamoy/aria2p/src/aria2p/api.py", line 521, in pause
    pause_func(download.gid)
    │          │        └ <property object at 0x7fed038783b8>
    │          └ <aria2p.downloads.Download object at 0x7fecfa6bd240>
    └ <bound method Client.pause of Client(host='http://localhost', port=15449, secret='********')>

  File "/media/data/dev/pawamoy/aria2p/src/aria2p/client.py", line 672, in pause
    return self.call(self.PAUSE, [gid])
           │    │    │    │       └ '0000000000000002'
           │    │    │    └ 'aria2.pause'
           │    │    └ Client(host='http://localhost', port=15449, secret='********')
           │    └ <function Client.call at 0x7fed03971c80>
           └ Client(host='http://localhost', port=15449, secret='********')

  File "/media/data/dev/pawamoy/aria2p/src/aria2p/client.py", line 261, in call
    return self.res_or_raise(self.post(self.get_payload(method, params, msg_id=msg_id)))
           │    │            │    │    │    │           │       │              └ None
           │    │            │    │    │    │           │       └ ['0000000000000002']
           │    │            │    │    │    │           └ 'aria2.pause'
           │    │            │    │    │    └ <staticmethod object at 0x7fed0394c630>
           │    │            │    │    └ Client(host='http://localhost', port=15449, secret='********')
           │    │            │    └ <function Client.post at 0x7fed03971ea0>
           │    │            └ Client(host='http://localhost', port=15449, secret='********')
           │    └ <staticmethod object at 0x7fed0394c6a0>
           └ Client(host='http://localhost', port=15449, secret='********')

  File "/media/data/dev/pawamoy/aria2p/src/aria2p/client.py", line 391, in res_or_raise
    raise Client.response_as_exception(response)
          │      │                     └ {'id': -1, 'jsonrpc': '2.0', 'error': {'code': 1, 'message': 'GID#0000000000000002 cannot be paused now'}}
          │      └ <staticmethod object at 0x7fed039877b8>
          └ <class 'aria2p.client.Client'>

aria2p.client.ClientException: GID#0000000000000002 cannot be paused now
